### PR TITLE
FE-791 - Adds Cypress integration tests for the navigation

### PR DESCRIPTION
**Pull request by nicklemmon**
[FE-791](https://jira.int.messagesystems.com/browse/FE-791)

### What Changed
- Adds Cypress integration tests for the navigation - some already existed, however, adding support for dynamic rendering of navigation items based on fixtures
- Fixed a templates test that was demonstrating some flakiness

### How To Test
- Verify tests are passing
- Verify tests are testing relevant scenarios controlled by user/account data - **I am probably missing some scenarios!**

### To Do
- [ ] Incorporate feedback

### CODE REVIEW

File: cypress/support/commands.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ Cypress.Commands.add('stubAuth', () => {
     url: '/api/v1/authenticate',
     status: 200,
     response: '@authenticatePost',
-  }).as('stubbedTwoFactorRequest');
+  }).as('stubbedAuthenticateRequest');
```
Comment (nicklemmon) [inline cypress/support/commands.js]: This was obviously just mislabelled.

File: cypress/fixtures/account/200.get.does-not-have-inbox-placement.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      "plan_volume": 50000,
+      "period": "month",
+      "recurring_charge": 20,
+      "overage": 1,
+      "self_serve": true,
+      "type": "default"
+    },
+    "options": {
+      "ui": {},
+      "inbox_placement": false
```
Comment (bkemper) [inline cypress/fixtures/account/200.get.does-not-have-inbox-placement.json]: Where can maintain a list of Cypress ideas/future improvements?

Comment (nicklemmon) [reply cypress/fixtures/account/200.get.does-not-have-inbox-placement.json]: Hmm, a few ways to do it - could add it to the Wiki somewhere, or maybe we have a `TODO.md` inside the Cypress folder

**Comment by nicklemmon** [comment]
Closing while making some changes to fixtures.

File: cypress/tests/integration/navigation/navigation.spec.js [hunk 3]
```diff
@@ -48,39 +34,323 @@ describe('Mobile Navigation', () => {
...
+          .should('be.visible')
+          .closest('a')
+          .should('have.attr', 'href', '/reports/message-events');
+
+        cy.queryByText('Content').should('be.visible');
+        cy.queryByText('Recipients').should('be.visible');
+
+        cy.queryByText('Alerts')
+          .should('be.visible')
+          .should('be.visible')
```
Comment (bkemper) [inline cypress/tests/integration/navigation/navigation.spec.js]: This is really visible ✔️

File: cypress/tests/integration/navigation/navigation.spec.js [hunk 4]
```diff
@@ -48,39 +34,323 @@ describe('Mobile Navigation', () => {
...
+    it('renders navigation items intended for "admin" users', () => {
+      stubGrantsRequest({ role: 'admin' });
+
+      cy.get(navigationListSelector).within(() => {
+        cy.queryByText('Dashboard')
+          .should('be.visible')
+          .closest('a')
+          .should('have.attr', 'href', '/dashboard');
+
+        cy.queryByText('Signals Analytics').should('be.visible');
```
Comment (bkemper) [inline cypress/tests/integration/navigation/navigation.spec.js]: Should we or can we test the contents of this menu?

File: cypress/tests/integration/navigation/navigation.spec.js [hunk 5]
```diff
@@ -48,39 +34,323 @@ describe('Mobile Navigation', () => {
...
+          .should('have.attr', 'href', 'https://developers.sparkpost.com/api');
+
+        cy.queryByText('Log Out')
+          .should('be.visible')
+          .should('have.attr', 'href', '/logout');
+      });
+    });
+
+    it('renders navigation items intended only for "reporting" users', () => {
+      stubGrantsRequest({ role: 'reporting' });
```
Comment (bkemper) [inline cypress/tests/integration/navigation/navigation.spec.js]: I think we are missing tests for the "subaccount_reporting" role.

File: cypress/tests/integration/navigation/navigation.spec.js [hunk 6]
```diff
@@ -48,39 +34,323 @@ describe('Mobile Navigation', () => {
...
+      openAccountMenu();
+
+      cy.get(accountDropdownListSelector).within(() => {
+        cy.queryByText('Profile')
+          .should('be.visible')
+          .should('have.attr', 'href', '/account/profile');
+
+        cy.queryByText('API Docs')
+          .should('be.visible')
+          .should('have.attr', 'href', 'https://developers.sparkpost.com/api');
```
Comment (bkemper) [inline cypress/tests/integration/navigation/navigation.spec.js]: Is it important to test the href?

**Pull request closed without merging**